### DISCUSSION TITLE
New Project Wizard, Soft Scene Breaks, Project Metadata and Pandoc

### DISCUSSION THREAD

**Discussion by awqk** [vkbo/novelWriter#1701] [Ideas]
I installed NW only yesterday, and I am importing an existing project now. So far everything looks very nice, neat, useful and stable. Thank you!

Up to now, I have used a variety of code editors to write text in Markdown. In the build step I create a manuscript file using the industry-standard [Shunn-Format]( https://www.shunn.net/format/). This is done with pandoc, bash, and a .docx template. It looks like I won't be able to get rid of the final build step, but I might get rid of the Spreadsheet tracking the scene order, plot, timelines and characters.

Suggestions:

-  allow creating a new project in an existing directory. This should be OK if the .nwx file does not yet exist.

-  soft scene breaks: this appear as sections in the Outline view. How about a modifier for soft scene breaks
  "###! Soft Scene Break", similar to "#! Title" and "##! Chapter"

Question:
- what would be the best way to insert a YAML metadata block or a pandoc title block (both pandoc extensions) into the md build output, to set author, title etc. Sure, I can do this in my final build shell script, but there may be nicer ways to do this.

Perhaps I should have started 3 topics...

**Comment by vkbo** [comment]
> allow creating a new project in an existing directory. This should be OK if the .nwx file does not yet exist.

This is restricted for a reason. novelWriter needs to be in a folder by itself, and you should never mix other files into it. Consider the novelWriter project folder a "file". The backup of the project will also ignore any file that doesn't belong there, and there is a single file project format in the works as well.

>  soft scene breaks: this appear as sections in the Outline view. How about a modifier for soft scene breaks 
>       "###! Soft Scene Break", similar to "#! Title" and "##! Chapter"

Sections are *not* soft scene breaks. If you use them as such, the novel structure does not make fully sense. There is already a feature proposal for a soft scene notation, but the problem with a soft scene break is that whether it is "soft" or "hard" depends on the preceding scene, so it may not hold if you move them around.

In any case, the feature ticket is here: #1050 

> what would be the best way to insert a YAML metadata block or a pandoc title block (both pandoc extensions) into the md build output, to set author, title etc. Sure, I can do this in my final build shell script, but there may be nicer ways to do this.

This is out of scope. If you want to post process your manuscript in scripts, or do more advanced things, you can use the JSON wrapped Markdown or HTML output.

**Reply by vkbo** [reply]
As for the cover page, no matter how many claim there is a standard, there isn't a universal one, so I have generally dodged the issue by not generating cover pages for manuscripts. Markdown formatting is limited, but some of the standards you can reproduce by creating a single document at the top of the project. Others are harder to reproduce.

I haven't decided what exactly to do about it, or if anything should be done. I'm considering adding word stats options to the manuscript tool that could potentially be inserted into a generated cover page.

**Reply by awqk** [reply]
> > allow creating a new project in an existing directory. This should be OK if the .nwx file does not yet exist.

> This is restricted for a reason.  [...]

I was thinking about continuing with the same folder where my git repository already resides, given that git handles renames rather well (moving the original content to a different file in the content folder), but probably I am better off starting a new repository. I am trying to do things the right way here, because I will have to live with my decision for quite some time.

> > soft scene breaks: this appear as sections in the Outline view. [...]

> Sections are _not_ soft scene breaks. If you use them as such, the novel structure does not make fully sense. [...]

This is what the manual recommends at the moment, but it is good to see the issue is already known.

> As for the cover page [...] I have generally dodged the issue [...]

I can relate to that, and it is certainly not a big issue for me. I have a viable solution using template files and pandoc.

Thanks for your speedy clarifications, and keep up the good work!

**Reply by vkbo** [reply]
> > Sections are _not_ soft scene breaks. If you use them as such, the novel structure does not make fully sense. [...]
> 
> This is what the manual recommends at the moment, but it is good to see the issue is already known.

It is suggested as a workaround for the missing feature. But as the documentation itself clearly states, this is not what section headings are for.

> I can relate to that, and it is certainly not a big issue for me. I have a viable solution using template files and pandoc.

Feel free to submit feature ideas, if you have any. It's how most things are added to novelWeriter. I have limited the scope of the manuscript tool though, as it would otherwise consume all of my time as there are countless ways it can be integrated with other software. I draw the line at producing a manuscript that at most needs a tweaked cover page.

> Thanks for your speedy clarifications, and keep up the good work!

Thanks!

**Reply by hifron** [reply]
So [Allow for templates](https://github.com/vkbo/novelWriter/issues/841) issue is already closed... but create new project from command line the same way GraphicalUserInterface used that is maybe solution to this maybe with some templates delivered.

**Reply by vkbo** [reply]
The issue is closed because it has been implemented. You can copy an existing project during new project creation.
